flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/Cassandra40ConnectorITCase.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.cassandra;
+
+import org.apache.flink.util.DockerImageVersions;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.testcontainers.containers.CassandraContainer;
+
+/** Class for IT cases for all Cassandra sinks tested on Cassandra 4.0.x . */
+public class Cassandra40ConnectorITCase extends CassandraConnectorBaseTest {
```

Comment:
I'm not sure if we really need tests for different cassandra versions.

Reply:
You mean we should test only the latest version ? The ticket states that we should support all the versions that Apache Cassandra supports (3.0, 4.0, 3.11), how can we be sure that we support them if we don't have these version specific tests?

Reply:
One could argue that since the driver says that it supports it, it should be good enough to test a single version.
We also don't test our Kafka connector against 10 different kafka versions; we rely on the compatibility guarantees provided by the client.

Reply:
This makes total sense to me considering that there is neither prod code difference nor dependencies difference. I can definitely remove all the sub-classes and put all the test code back to `CassandraConnectorITCase` with a static 4.0.3 container. This will make the test code way simpler.